adapters/repos/db/lsmkv/memtable_serialization.go

```diff
@@ -0,0 +1,189 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package lsmkv
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+	"path/filepath"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/weaviate/weaviate/adapters/repos/db/roaringset"
+	"github.com/weaviate/weaviate/adapters/repos/db/roaringsetrange"
+)
+
+const currentVersion uint16 = 1
+
+func SerializeMemtable(m *Memtable, w io.Writer) error {
```

Comment:
Some fuzzing tests would be great here - have a look at adapters/repos/db/lsmkv/segmentindex/quantile_keys_test.go::FuzzQuantileKeys

Reply:
adding a checksum may also be relevent here, and do recover from the associated wal if there is a mismatch